Subject: [PATCH] NuPlayerRenderer calls updateMaxTimeMedia to avoid mediatime error

Video progress bar stay at 0 if audio gets eos at the begining. So
video should call updateMaxTimeMedia after audio is eos.

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index b742762..a148b74 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -112,6 +112,7 @@
       mVideoLateByUs(0ll),
       mHasAudio(false),
       mHasVideo(false),
+      mAudioEOS(false),
       mNotifyCompleteAudio(false),
       mNotifyCompleteVideo(false),
       mSyncQueues(false),
@@ -1212,7 +1213,7 @@
                 realTimeUs = nowUs;
             }
         }
-        if (!mHasAudio) {
+        if (!mHasAudio || mAudioEOS) {
             // smooth out videos >= 10fps
             mMediaClock->updateMaxTimeMedia(mediaTimeUs + 100000);
         }
@@ -1356,6 +1357,9 @@
 }
 
 void NuPlayer::Renderer::notifyEOS(bool audio, status_t finalResult, int64_t delayUs) {
+    if (audio)
+        mAudioEOS = true;
+
     if (audio && delayUs > 0) {
         sp<AMessage> msg = new AMessage(kWhatEOS, this);
         msg->setInt32("audioEOSGeneration", mAudioEOSGeneration);
@@ -1386,6 +1390,7 @@
 
     if (audio) {
         mHasAudio = true;
+        mAudioEOS = false;
     } else {
         mHasVideo = true;
     }
diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h
index fe7f8fa..9d78df2 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.h
@@ -167,6 +167,7 @@
     int64_t mVideoLateByUs;
     bool mHasAudio;
     bool mHasVideo;
+    bool mAudioEOS;
 
     bool mNotifyCompleteAudio;
     bool mNotifyCompleteVideo;
